File: Listas, Colas y Pilas/Colas/ColasArreglos.cpp
#include <iostream>
using namespace std;

#define MAXSIZE 5
int queue[MAXSIZE];
int front = -1, rear = -1;

void insertar() {
    int elemento;
    cout << "\nIngrese el elemento: ";
    cin >> elemento;

    if (rear == MAXSIZE - 1) {
        cout << "\nDESBORDAMIENTO (OVERFLOW)\n";
        return;
    }
    if (front == -1 && rear == -1) {
        front = rear = 0;
    } else {
        rear++;
    }

    queue[rear] = elemento;
    cout << "\nElemento insertado correctamente.\n";
}

void eliminar() {
    if (front == -1 || front > rear) {
        cout << "\nSUBDESBORDAMIENTO (UNDERFLOW)\n";
        return;
    }

    int elemento = queue[front];
    if (front == rear) {
        front = rear = -1;
    } else {
        front++;
    }

    cout << "\nElemento eliminado: " << elemento << "\n";
}

void mostrar() {
    if (rear == -1 || front == -1 || front > rear) {
        cout << "\nLa cola está vacía.\n";
    } else {
        cout << "\nElementos en la cola:\n";
        for (int i = front; i <= rear; i++) {
            cout << queue[i] << "\n";
        }
    }
}

int main() {
    int opcion = 0;

    while (opcion != 4) {
        cout << "\n************** MENÚ PRINCIPAL ***************\n";
        cout << "1. Insertar un elemento\n";
        cout << "2. Eliminar un elemento\n";
        cout << "3. Mostrar la cola\n";
        cout << "4. Salir\n";
        cout << "Ingrese su opción: ";
        cin >> opcion;

        switch (opcion) {
        case 1: insertar(); break;
        case 2: eliminar(); break;
        case 3: mostrar(); break;
        case 4: cout << "\nSaliendo del programa...\n"; break;
        default: cout << "\nOpción inválida.\n"; break;
        }
    }

    return 0;
}

File: Listas, Colas y Pilas/Colas/ColasArreglos.cs
using System;

class ColaArreglo {
    const int MAXSIZE = 5;
    static int[] queue = new int[MAXSIZE];
    static int front = -1, rear = -1;

    static void Insertar() {
        Console.Write("\nIngrese el elemento: ");
        int elemento = int.Parse(Console.ReadLine());

        if (rear == MAXSIZE - 1) {
            Console.WriteLine("\nOVERFLOW");
            return;
        }

        if (front == -1 && rear == -1) {
            front = rear = 0;
        } else {
            rear++;
        }

        queue[rear] = elemento;
        Console.WriteLine("Elemento insertado.");
    }

    static void Eliminar() {
        if (front == -1 || front > rear) {
            Console.WriteLine("\nUNDERFLOW");
            return;
        }

        int elemento = queue[front];
        if (front == rear) front = rear = -1;
        else front++;

        Console.WriteLine($"Elemento eliminado: {elemento}");
    }

    static void Mostrar() {
        if (front == -1 || front > rear) {
            Console.WriteLine("La cola está vacía.");
            return;
        }

        Console.WriteLine("\nElementos en la cola:");
        for (int i = front; i <= rear; i++)
            Console.WriteLine(queue[i]);
    }

    static void Main(string[] args) {
        int opcion = 0;

        while (opcion != 4) {
            Console.WriteLine("\n1. Insertar\n2. Eliminar\n3. Mostrar\n4. Salir");
            Console.Write("Opción: ");
            opcion = int.Parse(Console.ReadLine());

            switch (opcion) {
                case 1: Insertar(); break;
                case 2: Eliminar(); break;
                case 3: Mostrar(); break;
            }
        }
    }
}
